support/cas-server-support-saml-idp-web/src/main/java/org/apereo/cas/support/saml/web/idp/profile/sso/SSOSamlIdPPostProfileHandlerEndpoint.java

```diff
@@ -95,22 +101,57 @@ public SSOSamlIdPPostProfileHandlerEndpoint(final CasConfigurationProperties cas
      */
     @GetMapping(produces = MediaType.APPLICATION_XML_VALUE)
     @ResponseBody
-    public ResponseEntity<Object> produce(final HttpServletRequest request, final HttpServletResponse response) {
+    public ResponseEntity<Object> produceGet(final HttpServletRequest request, final HttpServletResponse response) {
+
+        val username = request.getParameter("username");
+        val password = request.getParameter("password");
+        val entityId = request.getParameter(SamlProtocolConstants.PARAMETER_ENTITY_ID);
+        val encrypt = Boolean.parseBoolean(request.getParameter("encrypt"));
+        return produce(request, response, username, password, entityId, encrypt);
+    }
+
+    /**
+     * Produce response entity.
+     *
+     * @param request  the request
+     * @param response the response
+     * @param map the RequestBody
+     * @return the response entity
+     */
+    @PostMapping(produces = MediaType.APPLICATION_XML_VALUE)
+    @ResponseBody
+    public ResponseEntity<Object> producePost(final HttpServletRequest request,
+                                              final HttpServletResponse response,
+                                              final @RequestBody Map<String, String> map) {
+
+        val username = map.get("username");
+        val password = map.get("password");
+        val entityId = map.get(SamlProtocolConstants.PARAMETER_ENTITY_ID);
+        val encrypt = Boolean.parseBoolean(map.get("encrypt"));
+        return produce(request, response, username, password, entityId, encrypt);
+    }
+
+    private ResponseEntity<Object> produce(final HttpServletRequest request,
+                                           final HttpServletResponse response,
+                                           final String username,
+                                           final String password,
+                                           final String entityId,
+                                           final boolean encrypt) {
 
         try {
-            val username = request.getParameter("username");
-            val password = request.getParameter("password");
-            val entityId = request.getParameter(SamlProtocolConstants.PARAMETER_ENTITY_ID);
-            
             val selectedService = this.serviceFactory.createService(entityId);
             val registeredService = this.servicesManager.findServiceBy(selectedService, SamlRegisteredService.class);
             RegisteredServiceAccessStrategyUtils.ensureServiceAccessIsAllowed(registeredService);
 
+            val loadedService = (SamlRegisteredService) JSON_SERIALIZER.from(JSON_SERIALIZER.toString(registeredService));
```

Comment:
You already have a registered-service; why serialize it to JSON, only to deserialize it again, to get the object again?

Reply:
Setting the fields on the object directly from the ServicesManager persists the change in the stored object.  Serializing to a new object ensures that the functionality is not affected for actual requests for the service.

Reply:
I think I tried that, but clone() alone wasn't deep enough.  I can you point me to a util or something.  Serializing to new object seemed the cleanest.

Reply:
How about `BeanUtils.copyProperties(..., ...);` ?